File: go/emojiwords.go
package main

import (
	"fmt"
	"os"
	"os/exec"
)

func main() {

	word := os.Args[1]
	emoji := os.Args[2]
	gap := ":empty:"
	emo := fmt.Sprintf(":%s:", emoji)
	letterMap := mapCreation(gap, emo)

	var result string

	for i := 0; i < 5; i++ {
		for letter := range word {
			var dynamicGap string
			if(letter == len(word) - 1) {
				dynamicGap = ""
			} else {
				dynamicGap = gap
			}
			result = result + letterMap[string(word[letter])][i] + dynamicGap
		}
		if(i < 4) {
			result = result + "\n"
		}
	}

	copyToClipboard(result)
}

func mapCreation(gap string, emo string) map[string][5]string {
	letterMap := make(map[string][5]string)

	letterMap["a"] = [5]string{gap + emo + emo + gap, emo + gap + gap + emo, emo + emo + emo + emo, emo + gap + gap + emo, emo + gap + gap + emo}
	letterMap["b"] = [5]string{emo + emo + emo + gap, emo + gap + gap + emo, emo + emo + emo + emo, emo + gap + gap + emo, emo + emo + emo + gap}
	letterMap["c"] = [5]string{emo + emo + emo + emo, emo + gap + gap + gap, emo + gap + gap + gap, emo + gap + gap + gap, emo + emo + emo + emo}
	letterMap["d"] = [5]string{emo + emo + emo + gap, emo + gap + gap + emo, emo + gap + gap + emo, emo + gap + gap + emo, emo + emo + emo + gap}
	letterMap["e"] = [5]string{emo + emo + emo + emo, emo + gap + gap + gap, emo + emo + emo + emo, emo + gap + gap + gap, emo + emo + emo + emo}
	letterMap["f"] = [5]string{emo + emo + emo + emo, emo + gap + gap + gap, emo + emo + emo + emo, emo + gap + gap + gap, emo + gap + gap + gap}
	letterMap["g"] = [5]string{emo + emo + emo + emo, emo + gap + gap + gap, emo + gap + emo + emo, emo + gap + gap + emo, emo + emo + emo + emo}
	letterMap["h"] = [5]string{emo + gap + gap + emo, emo + gap + gap + emo, emo + emo + emo + emo, emo + gap + gap + emo, emo + gap + gap + emo}
	letterMap["i"] = [5]string{emo, emo, emo, emo, emo}
	letterMap["j"] = [5]string{emo + emo + emo + emo, gap + gap + emo + gap, gap + gap + emo + gap, emo + gap + emo + gap, emo + emo + emo + gap}
	letterMap["k"] = [5]string{emo + gap + gap + emo, emo + gap + emo + gap, emo + emo + gap + gap, emo + gap + emo + gap, emo + gap + gap + emo}
	letterMap["l"] = [5]string{emo + gap + gap + gap, emo + gap + gap + gap, emo + gap + gap + gap, emo + gap + gap + gap, emo + emo + emo + emo}
	letterMap["m"] = [5]string{emo + gap + gap + gap + emo, emo + emo + gap + emo + emo, emo + gap + emo + gap + emo, emo + gap + gap + gap + emo, emo + gap + gap + gap + emo}
	letterMap["n"] = [5]string{emo + gap + gap + gap + emo, emo + emo + gap + gap + emo, emo + gap + emo + gap + emo, emo + gap + gap + emo + emo, emo + gap + gap + gap + emo}
	letterMap["o"] = [5]string{emo + emo + emo + emo, emo + gap + gap + emo, emo + gap + gap + emo, emo + gap + gap + emo, emo + emo + emo + emo}
	letterMap["p"] = [5]string{emo + emo + emo + emo, emo + gap + gap + emo, emo + emo + emo + emo, emo + gap + gap + gap, emo + gap + gap + gap}
	letterMap["q"] = [5]string{emo + emo + emo + emo + emo, emo + gap + gap + gap + emo, emo + gap + gap + gap + emo, emo + gap + gap + emo + emo, emo + emo + emo + emo + emo}
	letterMap["r"] = [5]string{emo + emo + emo + emo, emo + gap + gap + emo, emo + emo + emo + emo, emo + gap + emo + gap, emo + gap + gap + emo}
	letterMap["s"] = [5]string{emo + emo + emo + emo, emo + gap + gap + gap, emo + emo + emo + emo, gap + gap + gap + emo, emo + emo + emo + emo}
	letterMap["t"] = [5]string{emo + emo + emo + emo + emo, gap + gap + emo + gap + gap, gap + gap + emo + gap + gap, gap + gap + emo + gap + gap, gap + gap + emo + gap + gap}
	letterMap["u"] = [5]string{emo + gap + gap + emo, emo + gap + gap + emo, emo + gap + gap + emo, emo + gap + gap + emo, emo + emo + emo + emo}
	letterMap["v"] = [5]string{emo + gap + gap + gap + emo, emo + gap + gap + gap + emo, emo + gap + gap + gap + emo, gap + emo + gap + emo + gap, gap + gap + emo + gap + gap}
	letterMap["w"] = [5]string{emo + gap + gap + gap + emo, emo + gap + gap + gap + emo, emo + gap + gap + gap + emo, emo + gap + emo + gap + emo, emo + emo + gap + emo + emo}
	letterMap["x"] = [5]string{emo + gap + gap + gap + emo, gap + emo + gap + emo + gap, gap + gap + emo + gap + gap, gap + emo + gap + emo + gap, emo + gap + gap + gap + emo}
	letterMap["y"] = [5]string{emo + gap + gap + gap + emo, gap + emo + gap + emo + gap, gap + gap + emo + gap + gap, gap + gap + emo + gap + gap, gap + gap + emo + gap + gap}
	letterMap["z"] = [5]string{emo + emo + emo + emo + emo, gap + gap + gap + emo + gap, gap + gap + emo + gap + gap, gap + emo + gap + gap + gap, emo + emo + emo + emo + emo}
	letterMap[" "] = [5]string{gap, gap, gap, gap, gap}

	return letterMap
}

func copyToClipboard(text string) error {
	copyCommand := exec.Command("pbcopy")
	in, err := copyCommand.StdinPipe()
	if err != nil {
		return err
	}

	if err := copyCommand.Start(); err != nil {
		return err
	}
	if _, err := in.Write([]byte(text)); err != nil {
		return err
	}
	if err := in.Close(); err != nil {
		return err
	}
	return copyCommand.Wait()
}

File: scala/emojiwords.scala
import java.awt.Toolkit
import java.awt.datatransfer.StringSelection

object EmojiWords extends App {

  val word = args(0)
  val emoji = args(1)
  val gap = ":empty:"
  val emo = s":$emoji:"

  val letterMap: Map[Char, Vector[String]] = Map(
    'a' -> Vector(s"$gap$emo$emo$gap", s"$emo$gap$gap$emo", s"$emo$emo$emo$emo", s"$emo$gap$gap$emo", s"$emo$gap$gap$emo"),
    'b' -> Vector(s"$emo$emo$emo$gap", s"$emo$gap$gap$emo", s"$emo$emo$emo$emo", s"$emo$gap$gap$emo", s"$emo$emo$emo$gap"),
    'c' -> Vector(s"$emo$emo$emo$emo", s"$emo$gap$gap$gap", s"$emo$gap$gap$gap", s"$emo$gap$gap$gap", s"$emo$emo$emo$emo"),
    'd' -> Vector(s"$emo$emo$emo$gap", s"$emo$gap$gap$emo", s"$emo$gap$gap$emo", s"$emo$gap$gap$emo", s"$emo$emo$emo$gap"),
    'e' -> Vector(s"$emo$emo$emo$emo", s"$emo$gap$gap$gap", s"$emo$emo$emo$emo", s"$emo$gap$gap$gap", s"$emo$emo$emo$emo"),
    'f' -> Vector(s"$emo$emo$emo$emo", s"$emo$gap$gap$gap", s"$emo$emo$emo$emo", s"$emo$gap$gap$gap", s"$emo$gap$gap$gap"),
    'g' -> Vector(s"$emo$emo$emo$emo", s"$emo$gap$gap$gap", s"$emo$gap$emo$emo", s"$emo$gap$gap$emo", s"$emo$emo$emo$emo"),
    'h' -> Vector(s"$emo$gap$gap$emo", s"$emo$gap$gap$emo", s"$emo$emo$emo$emo", s"$emo$gap$gap$emo", s"$emo$gap$gap$emo"),
    'i' -> Vector(emo, emo, emo, emo, emo),
    'j' -> Vector(s"$emo$emo$emo$emo", s"$gap$gap$emo$gap", s"$gap$gap$emo$gap", s"$emo$gap$emo$gap", s"$emo$emo$emo$gap"),
    'k' -> Vector(s"$emo$gap$gap$emo", s"$emo$gap$emo$gap", s"$emo$emo$gap$gap", s"$emo$gap$emo$gap", s"$emo$gap$gap$emo"),
    'l' -> Vector(s"$emo$gap$gap$gap", s"$emo$gap$gap$gap", s"$emo$gap$gap$gap", s"$emo$gap$gap$gap", s"$emo$emo$emo$emo"),
    'm' -> Vector(s"$emo$gap$gap$gap$emo", s"$emo$emo$gap$emo$emo", s"$emo$gap$emo$gap$emo", s"$emo$gap$gap$gap$emo", s"$emo$gap$gap$gap$emo"),
    'n' -> Vector(s"$emo$gap$gap$gap$emo", s"$emo$emo$gap$gap$emo", s"$emo$gap$emo$gap$emo", s"$emo$gap$gap$emo$emo", s"$emo$gap$gap$gap$emo"),
    'o' -> Vector(s"$emo$emo$emo$emo", s"$emo$gap$gap$emo", s"$emo$gap$gap$emo", s"$emo$gap$gap$emo", s"$emo$emo$emo$emo"),
    'p' -> Vector(s"$emo$emo$emo$emo", s"$emo$gap$gap$emo", s"$emo$emo$emo$emo", s"$emo$gap$gap$gap", s"$emo$gap$gap$gap"),
    'q' -> Vector(s"$emo$emo$emo$emo$emo", s"$emo$gap$gap$gap$emo", s"$emo$gap$gap$gap$emo", s"$emo$gap$gap$emo$emo", s"$emo$emo$emo$emo$emo"),
    'r' -> Vector(s"$emo$emo$emo$emo", s"$emo$gap$gap$emo", s"$emo$emo$emo$emo", s"$emo$gap$emo$gap", s"$emo$gap$gap$emo"),
    's' -> Vector(s"$emo$emo$emo$emo", s"$emo$gap$gap$gap", s"$emo$emo$emo$emo", s"$gap$gap$gap$emo", s"$emo$emo$emo$emo"),
    't' -> Vector(s"$emo$emo$emo$emo$emo", s"$gap$gap$emo$gap$gap", s"$gap$gap$emo$gap$gap", s"$gap$gap$emo$gap$gap", s"$gap$gap$emo$gap$gap"),
    'u' -> Vector(s"$emo$gap$gap$emo", s"$emo$gap$gap$emo", s"$emo$gap$gap$emo", s"$emo$gap$gap$emo", s"$emo$emo$emo$emo"),
    'v' -> Vector(s"$emo$gap$gap$gap$emo", s"$emo$gap$gap$gap$emo", s"$emo$gap$gap$gap$emo", s"$gap$emo$gap$emo$gap", s"$gap$gap$emo$gap$gap"),
    'w' -> Vector(s"$emo$gap$gap$gap$emo", s"$emo$gap$gap$gap$emo", s"$emo$gap$gap$gap$emo", s"$emo$gap$emo$gap$emo", s"$emo$emo$gap$emo$emo"),
    'x' -> Vector(s"$emo$gap$gap$gap$emo", s"$gap$emo$gap$emo$gap", s"$gap$gap$emo$gap$gap", s"$gap$emo$gap$emo$gap", s"$emo$gap$gap$gap$emo"),
    'y' -> Vector(s"$emo$gap$gap$gap$emo", s"$gap$emo$gap$emo$gap", s"$gap$gap$emo$gap$gap", s"$gap$gap$emo$gap$gap", s"$gap$gap$emo$gap$gap"),
    'z' -> Vector(s"$emo$emo$emo$emo$emo", s"$gap$gap$gap$emo$gap", s"$gap$gap$emo$gap$gap", s"$gap$emo$gap$gap$gap", s"$emo$emo$emo$emo$emo"),
    ' ' -> Vector(gap, gap, gap, gap, gap)
  )

  val result =
    (0 to 4).map { i =>
      word.map { letter =>
        letterMap(letter)(i) + gap
      }.mkString.dropRight(7) + "\n"
    }.mkString.dropRight(1)

  val clipboard = Toolkit.getDefaultToolkit().getSystemClipboard()
  val transferable = new StringSelection(result)
  clipboard.setContents(transferable, null)
}
